### PR TITLE
OCPBUGS-41175: Operator Hub does not show correct description when installing new operator

### PR DESCRIPTION
**Pull request by Mylanos**
Links to: https://issues.redhat.com/browse/OCPBUGS-41175 

Problem: The description of the operator hub item didnt show accordingly to the selected channel.

Fix: 
Pass the description from the `currentCSVDesc` to the `MarkdownView` instead of the longDescription, that is derived from the default channel.

Before:
Screenshot 2024-09-30 at 12 40 40

After 
Screenshot 2024-09-30 at 12 05 43

### CODE REVIEW

**Review by TheRealJon — APPROVED** [review]
/lgtm

**Comment by Mylanos** [comment]
After making the initial change that fixed the issue of not showing the correct description for a given channel, a problem of unstable styling within the MarkdownView popped up, which can be seen on the following screen recording.

https://github.com/user-attachments/assets/74770aaf-d1b8-4474-b70f-d6db1732e7f5

Cause of problem was that the `iframe` component the `MarkdownView` uses, was mounting/creating new `iframe` component each time the content/description has changed, which caused the styling to be reapplied over and over again ( the CSS styling is defined within the HTML page that is being generated in this component and passed to the `iframe` ). Which resulted in a problem where the styling was inconsistent - most notable issue was that the background color flashed to white. 

To resolve this issue I kept the `iframe` mounted and only made changes to its content based on the current description/content, while keeping the `onLoad` logic. The content is being injected into the `iframe` instead of using the `srcDoc`, every time it changes.

After these changes, the flickering problem does not occur.

https://github.com/user-attachments/assets/3772f8a2-970e-40de-bd9e-aa80ae3a8fc4

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-item-details.tsx [hunk 1]
```diff
@@ -219,11 +219,11 @@ export const OperatorHubItemDetails: React.FC<OperatorHubItemDetailsProps> = ({
   const {
     catalogSource,
     source,
+    longDescription,
```
Comment (rhamilto) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-item-details.tsx:222]: Nit:  unnecessary change adds noise

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-item-details.tsx [hunk 2]
```diff
@@ -254,6 +254,7 @@ export const OperatorHubItemDetails: React.FC<OperatorHubItemDetailsProps> = ({
   }, [item?.obj?.status, setDeprecatedPackage]);
   const currentChannel = obj?.status.channels.find((ch) => ch.name === installChannel);
   const selectedChannelContainerImage = currentChannel?.currentCSVDesc.annotations.containerImage;
+  const selectedChannelDescription = currentChannel?.currentCSVDesc.description ?? longDescription;
```
Comment (rhamilto) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-item-details.tsx:257]: If `currentChannel?.currentCSVDesc.description === ''`, it will display instead of `longDescription`.

File: frontend/packages/console-plugin-shared/src/components/Markdown/MarkdownView.tsx [hunk 3]
```diff
@@ -183,39 +184,30 @@ const IFrameMarkdownView: React.FC<InnerSyncMarkdownProps> = ({
...
+  const frameRef = React.useRef<HTMLIFrameElement>(null);
   const [frameHeight, setFrameHeight] = React.useState(0);
   const [loaded, setLoaded] = React.useState(false);
-  const htmlTagElement = frame?.contentDocument?.documentElement;
 
   // eslint-disable-next-line react-hooks/exhaustive-deps
   const updateDimensions = React.useCallback(
     _.debounce(() => {
-      if (!frame?.contentWindow?.document?.body?.firstElementChild) {
+      if (!frameRef?.current?.contentWindow) {
```
[the remark this answers is not in the document]
Comment (Mylanos) [reply frontend/packages/console-plugin-shared/src/components/Markdown/MarkdownView.tsx:194]: Here I might have to keep the second optional for the `current` as follows`!frameRef.current?.contentWindow`
One of the integration test was failing after making this change due to `reading contentWindow from undefined` error. 
I think this occurs when the iframe is not fully mounted yet ( frameRef.current is still null), so the `useResizeObserver` fallbacks to a `<body>` element instead of the iframe and the callback to `updateDimensions` tries to read `contentWindow`, which is undefined at that point.

**Comment by Mylanos** [comment]
Thanks for the review @vojtechszocs and @rhamilto, updated the PR with those recommended changes 👍

File: frontend/packages/console-plugin-shared/src/components/Markdown/MarkdownView.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (vojtechszocs) [reply frontend/packages/console-plugin-shared/src/components/Markdown/MarkdownView.tsx]: Agreed, in that case you could also change the code from
```ts
if (!frameRef.current?.contentWindow) {
  return;
}
setFrameHeight(
  frameRef.current.contentWindow.document.body.firstElementChild.scrollHeight + (exactHeight ? 0 : 15),
);
```
to
```ts
if (frameRef.current?.contentWindow) {
  setFrameHeight(
    frameRef.current.contentWindow.document.body.firstElementChild.scrollHeight + (exactHeight ? 0 : 15),
  );
}
```
